src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_field_operator.py

```diff
@@ -0,0 +1,133 @@
+from gt4py.next.program_processors.runners.dace_fieldview.gtir_dataflow_builder import (
+    GtirDataflowBuilder,
+)
+from gt4py.next.type_system import type_specifications as ts
+
+
+class GtirBuiltinAsFieldOp(GtirDataflowBuilder):
+    """Generates the dataflow subgraph for the `as_field_op` builtin function."""
+
+    _stencil_expr: itir.Lambda
+    _stencil_args: list[Callable]
+    _field_domain: dict[Dimension, tuple[str, str]]
+    _field_type: ts.FieldType
+
+    def __init__(
+        self,
+        sdfg: dace.SDFG,
+        state: dace.SDFGState,
+        data_types: dict[str, ts.FieldType | ts.ScalarType],
+        node: itir.FunCall,
+        stencil_args: list[Callable],
+    ):
+        super().__init__(sdfg, state, data_types)
+
+        assert cpm.is_call_to(node.fun, "as_fieldop")
+        assert len(node.fun.args) == 2
+        stencil_expr, domain_expr = node.fun.args
+        # expect stencil (represented as a lambda function) as first argument
+        assert isinstance(stencil_expr, itir.Lambda)
+        # the domain of the field operator is passed as second argument
+        assert isinstance(domain_expr, itir.FunCall)
+
+        # visit field domain
+        domain = self.visit_domain(domain_expr)
+
+        # add local storage to compute the field operator over the given domain
+        # TODO: use type inference to determine the result type
+        node_type = ts.ScalarType(kind=ts.ScalarKind.FLOAT64)
+
+        self._field_domain = {dim: (lb, ub) for dim, lb, ub in domain}
+        self._field_type = ts.FieldType([dim for dim, _, _ in domain], node_type)
+        self._stencil_expr = stencil_expr
+        self._stencil_args = stencil_args
+
+    def _build(self) -> list[tuple[dace.nodes.Node, ts.FieldType | ts.ScalarType]]:
+        # generate a tasklet node implementing the stencil function and represent
+        # the field operator as a mapped tasklet, which will range over the field domain
+        output_connector = "__out"
+        tlet_code = "{var} = {code}".format(
+            var=output_connector, code=self.visit_symbolic(self._stencil_expr.expr)
+        )
+
+        # allocate local temporary storage for the result field
+        field_shape = [
+            # diff between upper and lower bound
+            f"{self._field_domain[dim][1]} - {self._field_domain[dim][0]}"
+            for dim in self._field_type.dims
+        ]
+        field_node = self._add_local_storage(self._field_type, field_shape)
+
+        # create map range corresponding to the field operator domain
+        map_ranges = {
+            f"i_{dim.value}": f"{lb}:{ub}" for dim, (lb, ub) in self._field_domain.items()
+        }
+
+        input_nodes: dict[str, dace.nodes.AccessNode] = {}
+        input_memlets: dict[str, dace.Memlet] = {}
+        assert len(self._stencil_args) == len(self._stencil_expr.params)
+        for arg, param in zip(self._stencil_args, self._stencil_expr.params):
```

Comment:
Should this be called with `strict=True`? The same comment applies to all `zip()` calls in this PR